plugins/obs-ffmpeg/texture-amf.cpp

```diff
@@ -1294,9 +1295,15 @@ static bool amf_avc_init(void *data, obs_data_t *settings)
 	int64_t bf = obs_data_get_int(settings, "bf");
 
 	if (enc->bframes_supported) {
-		set_avc_property(enc, MAX_CONSECUTIVE_BPICTURES, 3);
+		set_avc_property(enc, MAX_CONSECUTIVE_BPICTURES, bf);
 		set_avc_property(enc, B_PIC_PATTERN, bf);
```

Comment:
Both are required to be set in order to use B-frames for AVC. `B_PIC_PATTERN` serves to set the number of B-frames while `MAX_CONSECUTIVE_BPICTURES` limits the maximum. `MAX_CONSECUTIVE_BPICTURES` will play an active role when used in combination with PA and AdaptiveMiniGOP where PA analyzes the input content and decides the picture type of each frame adaptively so the pattern is variable instead of a fixed number equal to `B_PIC_PATTERN` in each GOP. Yes, both the `MAX_CONSECUTIVE_BPICTURES` and `B_PIC_PATTERN` properties will be set to the same value, however, the number of B-frames will vary up to the specified maximum when PA and AdaptiveMiniGOP are enabled as well.